addons/purchase/models/account_invoice.py

```diff
@@ -177,11 +188,21 @@ def _anglo_saxon_purchase_move_lines(self, i_line, res):
                             for val_stock_move in valuation_stock_move:
                                 valuation_price_unit_total += abs(val_stock_move.price_unit) * val_stock_move.product_qty
                                 valuation_total_qty += val_stock_move.product_qty
+                            # in Stock Move price unit is in company_currency
                             valuation_price_unit = valuation_price_unit_total / valuation_total_qty
                             valuation_price_unit = i_line.product_id.uom_id._compute_price(valuation_price_unit, i_line.uom_id)
```

Comment:
This would be in company currency since `val_stock_move.price_unit` is in company currency, right? See:
https://github.com/odoo/odoo/blob/06cd3e55104427ca0531ba9e446f101b180e1755/addons/stock/models/stock_move.py#L111

That would be inconsistent with what you did above, no?